nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -403,6 +411,238 @@ async fn run() -> Result<()> {
+        .await?;
+    let mut tasks: Vec<AssignerTask> = Vec::new();
+    while let Some(task) = cursor.try_next().await? {
+        tasks.push(task);
+    }
+
+    let mut processing_count = 0;
+    let mut answers: HashMap<String, i32> = HashMap::new();
+    for task in tasks.iter() {
+        tracing::debug!("Worker id: {}", hex::encode(task.worker.to_bytes()));
+        if task.result.is_empty() && task.status == AssignerTaskStatus::Pending {
+            processing_count += 1;
+            break;
+        }
+        let result_identifier = match task.get_result_identifier() {
+            Ok(val) => {
+                val
+            }
+            Err(err) => {
+                return Err(anyhow!("Failed to find result identifier for task: {}", task.event_id));
+            }
+        };
+        if let Some(identifier) = result_identifier {
+            if !identifier.is_empty() {
+                match answers.entry(identifier) {
+                    Entry::Occupied(mut entry) => {
+                        entry.insert(entry.get() + 1);
+                    }
+                    Entry::Vacant(entry) => {
+                        entry.insert(1);
+                    }
+                }
+            }
+        }
+    }
+    if processing_count > 0 {
+        tracing::debug!("Waiting for more results for event id: {}", event_id);
+        return Ok(());
+    }
+    tracing::debug!("All results has been collected for event id: {}", event_id);
+    // max_count count number of results having same max count
+    let mut max_count = 0;
+    // max_answer is number of results having max answer
+    let mut max_answer = 0;
+    let mut answer: String = "".to_string();
+    for result in answers.keys() {
+        let count = answers.get(result).unwrap().to_owned();
+        if count == max_answer {
+            max_count += 1;
+        } else if count > max_answer {
+            max_count = 1;
+            max_answer = count;
+            answer = result.clone();
+        }
+    }
+    let mut tasks_to_update: Vec<AssignerTask> = vec![];
+    if max_count > 1 {
+        tracing::debug!("There are {} results having same count: {}. We need to ask assign same task to more workers.", max_count, max_answer);
+        // Now let's mark all tasks with results as RETRY
+        for task in tasks {
+            if !task.result.is_empty() {
+                tasks_to_update.push(AssignerTask {
+                    worker: task.worker.clone(),
+                    event_id: task.event_id.clone(),
+                    status: AssignerTaskStatus::Retry,
+                    created_at: task.created_at,
+                    expires_at: task.expires_at,
+                    result: task.result.clone(),
+                });
+            }
+        }
+    } else {
+        // Now let's award the winners
+        let mut winners: Vec<Sender> = vec![];
+        for task in tasks {
+            // Safe to unwrap as we have get all results identifiers before
```

Comment:
This is not true. What was done before was to return in case or error (see also my question on why we return if one is bad), but we still can have tasks whose identifier is `None`, and those tasks will panic with `unwrap`.